app/javascript/components/Greetings.js

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchGreeting } from '../redux/message/greetingSlice';
+
+const Greetings = () => {
+  const {
+    greeting, isFetched, isLoading, hasError,
+  } = useSelector((store) => store.greeting);
+
+  const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(fetchGreeting());
+    }, [dispatch]);
+
+
+  if (isLoading) return <h1 className="loading">... Is Loading</h1>;
+  if (hasError) return <h1 className="error">An error has ocured</h1>;
+
+  return <h1 className="greeting">{greeting}</h1>;
+};
+
```

Comment:
Hello @HAMMAS-SALEEM, my code is working and it is displaying messages when I use this url **http://localhost:3000/**, kindly explain how this error happens, because in my end it is working as expect.
 thanks.

Reply:
Ok dear if it is working well then you can leave this change and work on the other changes